### PR TITLE
router: add router stat for reset after downstream response

### PR DESCRIPTION
**Pull request by ramaraochavali**
*Description*: Adds `upstream_rq_error` stat for upstream clusters.
*Risk Level*: Low
*Testing*: Added tests
*Docs Changes*: Updated
*Release Notes*: Updated
Fixes https://github.com/envoyproxy/envoy/issues/5905

### CODE REVIEW

File: docs/root/configuration/cluster_manager/cluster_stats.rst [hunk 1]
```diff
@@ -61,6 +61,7 @@ Every cluster has a statistics tree rooted at *cluster.<name>.* with the followi
   upstream_cx_none_healthy, Counter, Total times connection not established due to no healthy hosts
   upstream_rq_total, Counter, Total requests
   upstream_rq_active, Gauge, Total active requests
+  upstream_rq_error, Counter, Total requests that were reset after downstream response has already started.
```
Comment (venilnoronha) [inline docs/root/configuration/cluster_manager/cluster_stats.rst]: s/has already/had

File: test/common/router/router_test.cc [hunk 2]
```diff
@@ -110,7 +110,9 @@ class RouterTestBase : public TestBase {
                  "rq_success {} does not match expected {}",
                  cm_.conn_pool_.host_->stats_store_.counter("rq_success").value(), success);
     }
-    if (error != cm_.conn_pool_.host_->stats_store_.counter("rq_error").value()) {
+    if (error != cm_.conn_pool_.host_->stats_store_.counter("rq_error").value() &&
+        error != cm_.thread_local_cluster_.cluster_.info_->stats_store_.counter("upstream_rq_error")
+                     .value()) {
```
Comment (venilnoronha) [inline test/common/router/router_test.cc]: Split this into a separate `if` condition with appropriate error message?

File: docs/root/configuration/cluster_manager/cluster_stats.rst (hunk 1 above)
Comment (mattklein123) [reply docs/root/configuration/cluster_manager/cluster_stats.rst]: I know the host stat is named this, and honestly I can't remember why it's named that way, but can we consider a more descriptive name for this? Also, this description is very specific to HTTP which makes me think it's not very upstream generic. Should this instead be a router stat? Thoughts?

Comment (ramaraochavali) [reply docs/root/configuration/cluster_manager/cluster_stats.rst]: I was about to respond to the other comment - I agree , I think calling this  `rq_error`  is not correct. Also for host it is incremented at multiple places like https://github.com/envoyproxy/envoy/blob/master/source/common/router/router.cc#L563 so it accounts for all errors and limiting this to just for this specific case and calling it `rq_error` seems incorrect.
Do we have router stats or are you suggesting to add a new stat hierarchy?

Comment (mattklein123) [reply docs/root/configuration/cluster_manager/cluster_stats.rst]: We already have router stats, take a look at the router code files. It should be easy to add.

Comment (ramaraochavali) [reply docs/root/configuration/cluster_manager/cluster_stats.rst]: Yeah. Got it. Will make changes to add it to router stats.

**Comment by ramaraochavali** [comment]
@mattklein123 changed this to router level stat. Please see if this what you are suggesting.

**Pull request merged**